src/java/org/apache/cassandra/index/sai/IndexContext.java

```diff
@@ -408,6 +411,12 @@ public ConcurrentMap<Memtable, MemtableIndex> getLiveMemtables()
     public boolean supports(Operator op)
     {
         if (op.isLike() || op == Operator.LIKE) return false;
+        if (isAnalyzed)
+        {
+            if (op == Operator.ANALYZER_MATCHES) return true;
+            // Analyzed columns store the indexed result and are unable to compute raw equality
+            if (op == Operator.EQ) return false;
+        }
```

Comment:
It looks like this will fall through to the rest of the code if it's a different op which is probably not what we want

Reply:
Great point, we can go with `if (isAnalyzed) return op == Operator.ANALYZER_MATCHES;`